app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -365,17 +365,132 @@ protected function getCorrectedTotal($projectedTotal, $magentoDerivedCartData)
         return $magentoDerivedCartData;
     }
 
+    /**
+     * Corrects the address format by assuring no lines exceed 50 characters.
+     * This character restriction seems to affect several merchants
+     *
+     * @param array $streetAddress     The street address to potentially correct
+     *
+     * @return bool true if a correction was made, otherwise false
+     *
+     * @throws Mage_Core_Exception  thrown if there is no natural to correct an erroneous address
+     */
+    public function correctStreetAddress(&$streetAddress) {
+        $originalStreetAddress = $streetAddress;
+        $this->reallocateAddressLines($streetAddress);
+        return $streetAddress === $originalStreetAddress;
+    }
+
+    /**
+     * If any address line is longer than 50, this adjust the array by shifting or concatenating
+     * the lines to accommodate.  If the fourth line is reached and the problem has not been
+     * remedied, the tail end of that address line is truncated to satisfy the restraint
+     *
+     * @param array $streetAddressArray   The street address data to be adjusted
+     * @param int $startingIndex
+     *
+     * @throws Mage_Core_Exception  thrown if there is no natural place break up the address,
+     */
+    private function reallocateAddressLines(&$streetAddressArray, $startingIndex=0){
+
+        /** @var Bolt_Boltpay_Helper_Bugsnag $bugsnag */
+        $bugsnag = Mage::helper('boltpay/bugsnag');
+
+        $startingIndex = min($startingIndex, 3);
+
+        ///////////////////////////////////////////
+        /// Make sure we have an array of length 4
+        ///////////////////////////////////////////
+        while (count($streetAddressArray) < 4) $streetAddressArray[] = '';
+        ///////////////////////////////////////////
+
+        $tempLine = $streetAddressArray[$startingIndex] = preg_replace('/\h+/', ' ', $streetAddressArray[$startingIndex]);
+
+        if ( strlen($tempLine) > 50 ) {
+            // The length of this line is too long.  We must try to correct it
+
+            if ( $startingIndex >= 3) {
+                // Since this is address line 4, the last available line, we will truncate the string
+                // to meet requirements and send a warning to Bugsnag
+                $streetAddressArray[$startingIndex] = substr($tempLine, 0, 50);
+
+                $message = "Cropped address line '$tempLine' to '".$streetAddressArray[$startingIndex]."'\nOriginal Address: ".var_export($streetAddressArray, true);
+                $bugsnag->notifyException(new Exception($message), array(), 'warning');
+
+                return;
+            }
+
+            ///////////////////////////////////////////////////////////
+            /// Try to find a natural splitting point in address line,
+            /// then split the string
+            ///////////////////////////////////////////////////////////
+            $breakPoint = strrpos(substr($tempLine, 0, 50), ' ');
+
```

Comment:
To cover the edge case where 51st character is space (exactly 50 remain for the line) you should change 50 to 51.